Autopilot/Src/gps.cpp

```diff
@@ -0,0 +1,386 @@
+/**
+ * Gps sensor
+ * Author(s): Dhruv Rawat
+ */
+
+#include "gps.hpp"
+#include "UART.hpp"
+#include "../Libraries/STM32F7xx_HAL_Driver/Src/stm32f7xx_hal_uart.c"
+#include <stddef.h>
+#include <string.h>
+
+
+#define KTS_TO_MPS 0.51444f //Converts from cm/s to m/s
+
+#define GPS_UART_BAUDRATE 9600
+#define GPS_UART_BUFFER_SIZE 100
+
+#define PUBX_CONFIG_NMEA "$PUBX,41,1,07,03,9600,0*25\r\n" //Change hex numbers later
+#define PUBX_SET_GGA "$PUBX,40,GGA,0,1,0,0,0,0*5D\r\n"
+#define PUBX_SET_VTG "$PUBX,40,VTG,0,1,0,0,0,0*5D\r\n"
+#define PUBX_SET_GGA_OFF "$PUBX,40,GGA,0,0,0,0,0,0*5D\r\n"
+#define PUBX_SET_VTG_OFF "$PUBX,40,VTG,0,0,0,0,0,0*5D\r\n"
+
+#define GGA_MESSAGE "GPGGA"
+#define VTG_MESSAGE "GPVTG"
+
+NEOM8* NEOM8::gps_Instance = NULL;
+UARTPort *gpsUART;
+
+char vtgValues[GPS_UART_BUFFER_SIZE];
+char ggaValues[GPS_UART_BUFFER_SIZE];
+char generalBuffer[GPS_UART_BUFFER_SIZE];
```

Comment:
I'd be much better to have these variables be private member variables than be global like this.

If you really want them out here though, make sure they're declared static. That way, they're file scope instead of global.

Reply:
Will keep them as static so I can access them in the handle_DMA_input() function.